main/src/ca/uwaterloo/flix/language/phase/Deriver.scala

```diff
@@ -167,6 +170,121 @@ object Deriver extends Phase[KindedAst.Root, KindedAst.Root] {
       KindedAst.MatchRule(pat, guard, exp)
   }
 
+  /**
+    * Creates a toString instance for the given enum.
+    *
+    * {{{
+    * enum E[a] with Eq {
+    *   case C1
+    *   case C2(a)
+    *   case C3(a, a)
+    * }
+    * }}}
+    *
+    * yields
+    *
+    * {{{
+    * instance Eq[E[a]] with Eq[a] {
+    *   pub def eq(x: E[a], y: E[a]): Bool = (x, y) match {
+    *     case (C0, C0) => true
+    *     case (C1(x0), C1(y0)) => x0 == y0
+    *     case (C2(x0, x1), C2(y0, y1)) => x0 == y0 and x1 == y1
+    *     case _ => false
+    *   }
+    * }
+    * }}}
+    */
+  private def createEq(enum: KindedAst.Enum, loc: SourceLocation, root: KindedAst.Root)(implicit flix: Flix): KindedAst.Instance = enum match {
```

Comment:
Usually all of these functions are called `mkX`. (You even have an mk later! :))